Ability to send emojis and images

vishwamartur opened the pull request:
Related to #12

Add functionality to send emojis and images in chat messages.

* **Schema Changes**:
  - Add `imageUrl` and `emoji` fields to the `Message` model in `prisma/schema.prisma`.
  - Update the `Message` model to handle optional `text`, `imageUrl`, and `emoji` fields.
  - Add `imageUrl` and `emoji` columns to the `Message` table in `prisma/migrations/20241108180915_user_message_updated/migration.sql`.

* **API Changes**:
  - Update the `Message` interface to include `imageUrl` and `emoji` fields in `src/app/api/messages/route.ts`.
  - Modify the `POST` function to handle `imageUrl` and `emoji` fields.
  - Update the `GET` function to return `imageUrl` and `emoji` fields.

* **UI Changes**:
  - Add an emoji picker and an image upload button to the input area in `src/app/components/center/CenterBottom.tsx`.
  - Update the `handleSubmit` function to include `imageUrl` and `emoji` fields.
  - Update the `SingleMessage` component to display emojis and images in `src/app/components/center/SingleMessage.tsx`.
  - Implement the `ImageComp` component to handle image uploads and previews in `src/app/utilcomponents/ImageComp.tsx`.

vishwamartur committed:
Ability to send emojis and images

Related to #12

Add functionality to send emojis and images in chat messages.

* **Schema Changes**:
  - Add `imageUrl` and `emoji` fields to the `Message` model in `prisma/schema.prisma`.
  - Update the `Message` model to handle optional `text`, `imageUrl`, and `emoji` fields.
  - Add `imageUrl` and `emoji` columns to the `Message` table in `prisma/migrations/20241108180915_user_message_updated/migration.sql`.

* **API Changes**:
  - Update the `Message` interface to include `imageUrl` and `emoji` fields in `src/app/api/messages/route.ts`.
  - Modify the `POST` function to handle `imageUrl` and `emoji` fields.
  - Update the `GET` function to return `imageUrl` and `emoji` fields.

* **UI Changes**:
  - Add an emoji picker and an image upload button to the input area in `src/app/components/center/CenterBottom.tsx`.
  - Update the `handleSubmit` function to include `imageUrl` and `emoji` fields.
  - Update the `SingleMessage` component to display emojis and images in `src/app/components/center/SingleMessage.tsx`.
  - Implement the `ImageComp` component to handle image uploads and previews in `src/app/utilcomponents/ImageComp.tsx`.

geekyoswald commented:
Hi, thanks for the quick contribution but the assessment will begin once the complete video related to the project is out and the same is announced on LinkedIn as well . Kindly wait till then

geekyoswald commented:
Hi @vishwamartur , great attempt !! 

But kindly solve the few problems that persist in your code. 

1. Kindly include your package.json file as well in the commit . 
2. Make sure typescript is not throwing errors. 
3. The emojis are not getting sent and the emoji box also is not closing.
4. The image is not getting uploaded.
5. There is no definition regarding were the uploaded images be kept and retrieved from ? 
6. The message sending functionality also seems to have few errors as it is not working.

vishwamartur committed:
Update `Message` interface to include `imageUrl` and `emoji` fields

* **Form Data Handling**
  - Replace destructuring assignment with form data extraction
  - Extract `senderId`, `receiverId`, `text`, `imageUrl`, and `emoji` from form data

* **Message Creation**
  - Include `imageUrl` and `emoji` fields in the `data` object for message creation

vishwamartur committed:
Add emoji picker and image upload input to CenterBottom component

* **Emoji Picker**
  - Add `setShowEmojiPicker(false)` to `handleEmojiClick` to close the emoji picker after selection

* **Image Upload**
  - Add `emoji-picker-react` dependency to `package.json`

vishwamartur committed:
updates

updates

vishwamartur commented:
<img width="1347" alt="Screenshot 2024-12-08 at 9 20 54 PM" src="https://github.com/user-attachments/assets/ae580074-cd8c-4415-be6c-ad41b18ab6af">

vishwamartur commented:
Hi @geekyoswald 
Please show me the errors you're encountering so that I can help resolve them. If possible, we can schedule a call to discuss the issues.
